include/lug/System/Path.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <string>
+#include <iostream>
+#include <lug/System/Export.hpp>
+
+#define MAX_PATH_SIZE 4096
+
+namespace lug {
+namespace System {
+
+namespace priv {
+    class PathImpl;
+} // namespace priv
+
+class LUG_PATH_API Path
+{
+public:
+    ~Path();
+    std::string getRoot() const;
+    std::string getHome() const;
+    std::string getCwd() const;
+    std::string getSave() const;
+private:
+    Path();

```

Comment:
Move the private constructor above the public section, to reflect how it was done in `Window`